clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1534,14 +1549,21 @@ void cir::FuncOp::print(OpAsmPrinter &p) {
       p, *this,
       {getSymVisibilityAttrName(), getAliaseeAttrName(),
        getFunctionTypeAttrName(), getLinkageAttrName(), getBuiltinAttrName(),
-       getNoProtoAttrName()});
+       getNoProtoAttrName(), getExtraAttrsAttrName()});
+
 
   if (auto aliaseeName = getAliasee()) {
     p << " alias(";
     p.printSymbolName(*aliaseeName);
     p << ")";
   }
 
+  if (!getExtraAttrs().getElements().empty()) {
+    p << " extra(";
+    p.printOptionalAttrDict(getExtraAttrs().getElements().getValue());
+    p << " )";
```

Comment:
Why the extra space here?

Reply:
Good question. It's because I wanted to use the build-in printer `printOptionalAttrDict` which always prints a blank right after the left brace:

https://github.com/llvm/clangir/blob/c1fa60b4cde512964544ab66404dea79dbc5dcb4/mlir/lib/IR/AsmPrinter.cpp#L2561 

If we don't want the blank we may need to duplicate the printer code here.

Reply:
It's fine for now, we can come back for this if we think we should at some point.